### PR TITLE
Add table layout dropdown

### PR DESCRIPTION
**Pull request by fti-smaske**
**Description**
This PR consists new feature Layout selection dropdown.

 Zeplin - https://app.zeplin.io/project/63060ed35bfc8011024a9a9a/screen/63218de8f938147aa6e0ec82
 
image

This feature is required in Traffic Counter Cloud application.

**Background context**

In this fetaure user can change layout of the page like Grid, table, charts etc. This dropdown will use a callback function to retrieve a string value. It also has a dropdown for Item per page in case some page settings are requires.

**props**
```
  disabled?: boolean
  onToggleOpenCallback?: (isOpen: boolean) => void
  onCloseCallback?: () => void;
  pageSizeOptions?: number[],
  onPageSizeChange: (size: number) => void,
  defaultPage?: number,
  defaultPageSize?: number,
  getLayout: (layout: string) => void,
  layoutText?: string,
  pageSizeText?: string,
  contentArray: IOptionsItem[];
  minWidth?: number;
  minHeight?: number;
  hasPageSettings?: boolean;
```

**Considerations on Implementation**
Background color of dropdown is kept consistance with other UI-kit dropdown

Here are some screenshots of the expected result.

**Storybook screenshot**
image
image

**Local expected Result screenshot**
image
image

### CODE REVIEW

File: src/Filters/molecules/TableLayoutToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+  border-top-right-radius: 3px;
+`;
+
+const InnerBox = styled.div`
+  border-bottom-left-radius: 3px;
+  border-bottom-right-radius: 3px;
+  box-shadow: 0 4px 9px 0 hsla(204, 22%, 67%, 0.07);
+  border-right: solid 1px hsl(207, 16%, 86%);
+  border-bottom: solid 1px hsl(207, 16%, 86%);
+  border-left: solid 1px hsl(207, 16%, 86%);
```
Comment (fti-skumar) [inline src/Filters/molecules/TableLayoutToggle.tsx:117]:
```suggestion
  border: solid 1px hsl(207,16%,86%);
  border-top: none;
```
We can refactor it like this.

File: src/Filters/molecules/TableLayoutToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+  background-color: hsl(200, 23%, 97%);
+`;
+
+const LayoutGroup = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding-top: 8px;
+  padding-right: 8px;
+  padding-left: 10px;
```
Comment (fti-skumar) [inline src/Filters/molecules/TableLayoutToggle.tsx:126]:
```suggestion
  padding: 8px 8px 0px 10px;
```
Similarly here as well.

File: src/Filters/molecules/TableLayoutToggle.tsx [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+  width: 250px;
+`;
+
+const PaginationGroup = styled.div`
+  display: flex;
+  border-top: ${({ theme: { colors } }) => colors.divider} 1px solid;
+  padding-top: 8px;
+  padding-right: 20px;
+  padding-left: 10px;
```
Comment (fti-skumar) [inline src/Filters/molecules/TableLayoutToggle.tsx:135]:
```suggestion
  padding: 8px 20px 0px 10px;
```
Here as well.

**Comment by fti-skumar** [comment]
Can we align page size dropdown box with the top button, because in zeplin design, they are aligned.

align

File: src/Filters/molecules/TableLayoutToggle.tsx (hunk 1 above)
Comment (fti-smaske) [reply src/Filters/molecules/TableLayoutToggle.tsx:117]: Done

File: src/Filters/molecules/TableLayoutToggle.tsx (hunk 2 above)
Comment (fti-smaske) [reply src/Filters/molecules/TableLayoutToggle.tsx:126]: Done

File: src/Filters/molecules/TableLayoutToggle.tsx (hunk 3 above)
Comment (fti-smaske) [reply src/Filters/molecules/TableLayoutToggle.tsx:135]: Done

File: src/Filters/molecules/FilterLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+  defaultPageSize: number,
+  getLayout: (layout: string) => void,
+  layoutText?: string,
+  pageSizeText?: string,
+  icon1: string,
+  icon2: string,
+  icon3?: string,
+  buttonValue1: string,
+  buttonValue2: string,
+  buttonValue3?: string
```
Comment (fti-bcarpenter) [inline src/Filters/molecules/FilterLayout.tsx:183]: I think it is better to have layoutData as attribute as array

layoutData =[
{
        icon: 'Status',
        id: 'grid1',
        tooltipText: 'Device ID:'
      },
{
        icon: 'Info',
        id: 'chart1',
        tooltipText: 'Bar Chart:'
      }
]

Also add another attribute hasPageSetting to show pagination dropdown in this layout.

**Review by fti-bcarpenter — CHANGES_REQUESTED** [review]

**Comment by fti-skumar** [comment]
Thanks for the update, please also fix lint issue along with requested changes.

File: src/Filters/molecules/FilterLayout.tsx (hunk 4 above)
Comment (fti-smaske) [reply src/Filters/molecules/FilterLayout.tsx]: Done

**Review by fti-skumar — APPROVED** [review]
Thanks for the update, LGTM!

**Review by fti-bcarpenter — APPROVED** [review]

File: src/Filters/molecules/FilterLayout.tsx [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+    `};
+    ${openState.position === 'top-left' && `
+      top: 0;
+      right: 0;
+      transform: translateY(calc( -100% - 5px ));
+    `};
+    ${openState.position === 'top-right' && `
+      top: 0;
+      left: 0;
+      transform: translateY(calc( -100% - 5px ));
```
Comment (atomworks) [inline src/Filters/molecules/FilterLayout.tsx:95]: Don't these  
```
&& `
```
need to be 
```
&& css`
```
?

It seems to be working but I feel like best practice is to use that `css`. Please correct me if I am mistaken, I'm not sure if this changes as it is nested

File: src/Filters/molecules/FilterLayout.tsx [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+  box-shadow: 0 4px 9px 0 hsla(204, 22%, 67%, 0.07);
+  border: solid 1px hsl(207,16%,86%);
+  border-top: none;
+  background-color: hsl(200, 23%, 97%);
+`;
+
+const LayoutGroup = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding: 8px 8px 0 10px;
```
Comment (atomworks) [inline src/Filters/molecules/FilterLayout.tsx:123]: Can we change this to `4px 8px 4px 10px`? This balances the layout a little better

File: src/Filters/molecules/FilterLayout.tsx [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+  width: auto;
+`;
+
+const PaginationGroup = styled.div`
+  display: flex;
+  border-top: ${({ theme: { colors } }) => colors.divider} 1px solid;
+  padding: 8px 8px 0 10px;
```
Comment (atomworks) [inline src/Filters/molecules/FilterLayout.tsx:130]: Can we change this on also to `4px 8px 4px 10px`?

File: storybook/src/stories/Filters/molecules/FilterLayout.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    }
+  ];
+
+  const onSizePerPageChange = useCallback((size: number) => {
+    setPageSize(size);
+  },[]);
+
+  const handleLayout = useCallback((layout: string) => {
+    
+  }, []);
```
Comment (atomworks) [inline storybook/src/stories/Filters/molecules/FilterLayout.stories.tsx:39]: For these, could change it so it uses the actions feature of Storybook and outputs some information to the actions tab there so we can see what content can be expected? 

There should be some other examples in Storybook you can reference. Perhaps the Standard Button is an easy one!

**Review by atomworks — CHANGES_REQUESTED** [review]
Few change requests but nothing major. 

Biggest thing is just adding some actions integration to the story.

File: src/Filters/molecules/FilterLayout.tsx (hunk 7 above)
Comment (fti-smaske) [reply src/Filters/molecules/FilterLayout.tsx]: Done

File: src/Filters/molecules/FilterLayout.tsx (hunk 6 above)
Comment (fti-smaske) [reply src/Filters/molecules/FilterLayout.tsx]: Done

File: storybook/src/stories/Filters/molecules/FilterLayout.stories.tsx (hunk 8 above)
Comment (fti-smaske) [reply storybook/src/stories/Filters/molecules/FilterLayout.stories.tsx]: Done

File: src/Filters/molecules/FilterLayout.tsx (hunk 5 above)
Comment (fti-smaske) [reply src/Filters/molecules/FilterLayout.tsx]: Done

**Review by atomworks — APPROVED** [review]

**Pull request merged**